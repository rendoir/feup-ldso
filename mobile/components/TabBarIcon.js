import React from 'react';
import { Icon } from 'native-base';

export default class TabBarIcon extends React.Component {
    render() {
        return (
            <Icon type='FontAwesome'
                name={this.props.name}
                size={40}
                style={{ marginBottom: -3, color: 'white' }}
            />
        );
    }
}
