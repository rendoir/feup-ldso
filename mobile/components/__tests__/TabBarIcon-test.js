import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TabBarIcon from '../TabBarIcon';

it('renders correctly focused', () => {
    const iconData = {"name": "Icon"};
    const tree = renderer.create(<TabBarIcon data={iconData} focused={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly unfocused', () => {
    const iconData = {"name": "Icon"};
    const tree = renderer.create(<TabBarIcon data={iconData} focused={false}/>).toJSON();

    expect(tree).toMatchSnapshot();
});
