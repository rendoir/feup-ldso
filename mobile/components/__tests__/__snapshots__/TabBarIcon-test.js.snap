// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly focused 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`renders correctly unfocused 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;
